go/vt/vtctl/workflow/materializer.go

```diff
@@ -299,12 +302,18 @@ func (mz *materializer) deploySchema() error {
 
 		var applyDDLs []string
 		for _, ts := range mz.ms.TableSettings {
-			if hasTargetTable[ts.TargetTable] {
-				// Table already exists.
+			if td := hasTargetTable[ts.TargetTable]; td != nil {
+				// Table already exists. Let's be sure that it doesn't already have data.
+				// We exclude multi-tenant migrations from this check as the target tables
+				// are expected to frequently have data from previously migrated tenants.
+				if !mz.IsMultiTenantMigration() && td.RowCount > 0 {
```

Comment:
`RowCount` is from `information_schema.tables` which may not always be updated. We have anecdotally found those stats not getting updated for significant periods of time: showing 0 even after several copy phases have run. On this branch I started a second workflow sharing tables with the first one and it passed this validation, failing with a`Duplicate Key` error.

Which was why we were going for the approach in this PR: https://github.com/vitessio/vitess/pull/16826/files#diff-d3a320c7b03791f5d24189e3ae6d7fcac814f4fa1b3d7c02d496b3d8f0adf588R1040-R1043.

Reply:
Yeah, the row count is an estimate. It can remain 0 for some time if you have not inserted/updated enough rows to go beyond the initial 16KiB page in the tablespace. I would guess that in your testing you only had a few very small rows in the table (e.g. `corder`). Doing multiple copy phases would not help or matter in this case either. I don't think that this is a very likely scenario production — since InnoDB updates the stats in information_schema when 1/16th of the table has changed, so now we're talking about *1KiB*, and that normally happens pretty quickly on a brand new table that starts out as 16KiB in size — but it's certainly possible (then you'd have to wait for one of the other things that trigger an update).

I can remove the "table has existing data" code here tin this PR hen if you prefer the other approach. That was not the primary focus in this PR anyway. Sound good?

Thanks!

Reply:
My tests were for a trivial number of rows, yes. But this was also reported by others recently for larger tables where the copy phase itself took minutes. They were using `Progress` and not seeing any changes there without `analyze table`  ... Not sure why the stats update didn't happen in that case. 

In any case, yes, let's handle the check for data in that other PR. Rest looks good.

Reply:
OK, I did that here: https://github.com/vitessio/vitess/pull/16810/commits/fee40da2b5e9b347d55fc45b98ad6e1717dee38a